astropy/io/fits/fitsrec.py

```diff
@@ -217,6 +218,9 @@ def __array_finalize__(self, obj):
         if obj is None:
             return
 
+        if isinstance(obj, FITS_rec):
+            self._character_as_bytes = obj._character_as_bytes
+
         if isinstance(obj, FITS_rec) and obj.dtype == self.dtype:
```

Comment:
If you indent this one level, it becomes part of the previous `isinstance(obj, FITS_rec)` and you can just do `if obj.dtype == self.dtype:`

Reply:
Ah but that changes the logic, because the ``else`` needs to get executed if ``isinstance(obj, FITS_rec) and not obj.dtype == self.dtype``

Reply:
Hmm, clearly a problem of not looking beyond my nose!